iceoryx_posh/source/roudi/port_manager.cpp

```diff
@@ -770,7 +770,17 @@ PortManager::acquirePublisherPortData(const capro::ServiceDescription& service,
                                       const RunnableName_t& runnable [[gnu::unused]], // @todo #25 Fix introspection
                                       const PortConfigInfo& portConfigInfo) noexcept
 {
-    /// @todo #25 Add unique check for single producer type - see acquireSenderPortData()
+    if (doesViolateCommunicationPolicy<iox::build::CommunicationPolicy>(service).and_then(
```

Comment:
this is really hard to read and quite unintuitive with 7 lines for the if condition. how about this
```
if(auto violatingProcessName = doesViolateCommunicationPolicy<iox::build::CommunicationPolicy>(service))
{
    // do stuff
}
```
or just move the call to `doesViolateCommunicationPolicy` out of the if condition

Reply:
Yes you are right, we have to improve this piece of code.

But I think the code snippet below would always be true since it is an assignment
> if(auto violatingProcessName = doesViolateCommunicationPolicy<iox::build::CommunicationPolicy>(service))
> {
>     // do stuff
> }